### PR TITLE
ci: Added CI Job to release Apple Silicon binary with the release GH Actions flow

### PR DESCRIPTION
**Pull request by jparrill**
## Description

This PR involves just a CI Job to be executed on release time.

- [x] Added new target `aarch64-apple-darwin` on `release.yml` which supports Apple Silicon processors from new M1/M2 Mac 

## Test
To test this pull request, you can run the following commands:

```shell
rustup target add aarch64-apple-darwin
cargo build --target=aarch64-apple-darwin --release
```

## Additional Information

I already validated that it works on a M1 processor

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

### CODE REVIEW

**Review by viccuad — APPROVED** [review]
LGTM, great contribution, thanks!

It would be a good moment to revisit if we want to create a release for this build too, in the `release` job.

**Comment by flavio** [comment]
Thanks for the contribution @jparrill !! Ask @viccuad said, it would be great to update the bottom part of the file you changed to ensure this new artifact is picked up when doing a release.

I don't have a machine to test the final binary, do you confirm this works fine?

**Comment by raulcabello** [comment]
Thanks for the contribution! I would add a new release step for `darwin-aarch64` . Otherwise we just build it, but nobody can download it.

**Comment by jparrill** [comment]
> Thanks for the contribution @jparrill !! Ask @viccuad said, it would be great to update the bottom part of the file you changed to ensure this new artifact is picked up when doing a release.
> 
> I don't have a machine to test the final binary, do you confirm this works fine?

Yep, I have the binary working on my machine:
image

**Comment by jparrill** [comment]
> Thanks for the contribution! I would add a new release step for `darwin-aarch64` . Otherwise we just build it, but nobody can download it. Edit: same as @flavio said. It looks like we were writing at the same time

Done, I'd force pushed the change ;)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -224,6 +225,17 @@ jobs:
...
+    - name: Download darwin-aarch64 binary
+      uses: actions/download-artifact@v2
+      with:
+        name: kwctl-darwin-aarch64
+
+    - name: Download darwin-aarch64 binary
+      uses: actions/download-artifact@v2
+      with:
+        name: kwctl-darwin-aarch64
+
```
Comment (flavio) [inline .github/workflows/release.yml:238]: This is repetition from above, can be dropped

**Review by jvanz — APPROVED** [review]

**Review by raulcabello — APPROVED** [review]

**Pull request merged**